static/js/settings_org.js

```diff
@@ -979,41 +945,67 @@ exports.build_page = function () {
         for (const [i, file] of Array.prototype.entries.call(file_input[0].files)) {
             form_data.append('file-' + i, file);
         }
-        if (night) {
-            error_field = $("#night-logo-section .realm-logo-file-input-error");
-            spinner = $("#night-logo-section .upload-logo-spinner");
-            button_text = $("#night-logo-section .upload-logo-button-text");
+        if (icon) {
+            error_field = $("#realm_icon_file_input_error");
+            error_field.hide();
+            spinner = $("#upload_icon_spinner").expectOne();
+            loading.make_indicator(spinner, {text: i18n.t("Uploading profile picture.")});
+            $("#upload_icon_button_text").expectOne().hide();
+
+            channel.post({
+                url: '/json/realm/icon',
+                data: form_data,
+                cache: false,
+                processData: false,
+                contentType: false,
+                success: function () {
+                    loading.destroy_indicator($("#upload_icon_spinner"));
+                    $("#upload_icon_button_text").expectOne().show();
+                },
+                error: function (xhr) {
+                    loading.destroy_indicator($("#upload_icon_spinner"));
+                    $("#upload_icon_button_text").expectOne().show();
+                    ui_report.error("", xhr, error_field);
+                },
+            });
```

Comment:
I am not sure but I think this can be further deduplicated.

Similar to variables used for differentiating between night and day mode, variables can be used for icon also and url can be written something like `'/json/realm/' + type` and then channel.post will be called only once.

As said, I am not sure whether this will be correct in terms of readability, maybe Tim can suggest whether we need to do this or not.

Reply:
Yeah, I thought of doing that but the handing `spinner` `button_text` `loading` `error_field`  `success` and `error` are different in all the cases (realm icon, day logo, night logo) that's why I did separately.   btw @sahil839 thanks for the review

Reply:
Sorry, but I am not understanding how these are different, for eg in case of success we are writing two statements- one to destroy indicator and one to show the button text for both cases of icon and logo. We can just do `loading.destroy_indicator(spinner)` and assign the value of spinner differently for all cases of icon, day logo and night logo.
Can you please elaborate more if I am missing something?

Reply:
@sahil839 yeah you are correct i just missed it Thanks a lot